src/init.cpp

```diff
@@ -364,26 +374,17 @@ ScopedInitializer::~ScopedInitializer() {
   finalize();
 }
 
-void initResourcePartitionerHandler(pika::resource::partitioner& rp,
+void initResourcePartitionerHandler(pika::resource::partitioner&,
                                     const pika::program_options::variables_map& vm) {
-  // Don't create the MPI pool if the user disabled it
+  namespace mpi = pika::mpi::experimental;
+  // Create the MPI pool if needed and unless the user disabled it
+  mpi::pool_create_mode pool_mode = mpi::pool_create_mode::pika_decides;
+  namespace mpi = pika::mpi::experimental;
   if (vm["dlaf:no-mpi-pool"].as<bool>())
-    return;
-
-  // Don't create the MPI pool if there is a single process
-  int ntasks;
-  DLAF_MPI_CHECK_ERROR(MPI_Comm_size(MPI_COMM_WORLD, &ntasks));
-  if (ntasks == 1)
-    return;
-
-  // Disable idle backoff on the MPI pool
-  using pika::threads::scheduler_mode;
-  auto mode = scheduler_mode::default_mode;
-  mode = scheduler_mode(mode & ~scheduler_mode::enable_idle_backoff);
-
-  // Create a thread pool with a single core that we will use for all
-  // communication related tasks
-  rp.create_thread_pool("mpi", pika::resource::scheduling_policy::static_priority, mode);
-  rp.add_resource(rp.numa_domains()[0].cores()[0].pus()[0], "mpi");
+    pool_mode = mpi::pool_create_mode::force_no_create;
+
+  namespace mpix = pika::mpi::experimental;
+  // create a pool for mpi if necessary
+  mpix::create_pool(mpix::get_pool_name(), pool_mode);
```

Comment:
What do you think about moving this logic completely to pika and not initializing the MPI pool through the resource partitioner callback? In the long term I think it makes little sense to have DLA-Future be responsible for creating the MPI pool if the rest of the MPI functionality is in pika as well. The difficulty is: how does pika know if the MPI pool is needed or not (in the generic case, i.e. if it's not running for DLA-Future)... In the ideal case doing the polling on the default pool would be as performant as a separate pool, then we wouldn't need to bother with this. Perhaps worth coming back to this once we know a bit more about performance on Grace-Hopper?